java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/processor/DefaultMessageProcessor.java

```diff
@@ -0,0 +1,65 @@
+package com.jashmore.sqs.processor;
+
+import com.amazonaws.annotation.ThreadSafe;
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.model.DeleteMessageResult;
+import com.amazonaws.services.sqs.model.Message;
+import com.jashmore.sqs.QueueProperties;
+import com.jashmore.sqs.argument.ArgumentResolutionException;
+import com.jashmore.sqs.argument.ArgumentResolverService;
+import com.jashmore.sqs.argument.acknowledge.Acknowledge;
+import lombok.AllArgsConstructor;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.concurrent.Future;
+
+/**
+ * Default implementation of the {@link MessageProcessor} that will simply resolve arguments, process the message and delete the
+ * message from the queue if it was completed successfully.
+ */
+@ThreadSafe
+@AllArgsConstructor
+public class DefaultMessageProcessor implements MessageProcessor {
+    private final ArgumentResolverService argumentResolverService;
+    private final QueueProperties queueProperties;
+    private final AmazonSQSAsync amazonSqs;
+    private final Method messageConsumerMethod;
+    private final Object messageConsumerBean;
+
+    @Override
+    public void processMessage(final Message message) throws MessageProcessingException {
+        final Parameter[] parameters = messageConsumerMethod.getParameters();
```

Comment:
I wanted to make it easy so that people could replace any part of the application if I write shit code or they need extra functionality. Therefore, this was how I allowed them to implement their own argument resolution :) So far it seems to work well!